src/it/scala/constellation/IntegrationSpec.scala

```diff
@@ -1,99 +1,103 @@
 package constellation
 
-import java.net.InetAddress
-
-import constellation.util.testing.FakeNode
-import org.constellation.NetworkLoadbalancer
-import org.constellation.primitives.node.{Addr, ErrorBody, Id, Info, NodeState}
-import org.scalatest.{BeforeAndAfterAll, FunSpec, Matchers}
-import cats.data.NonEmptyList
-import cats.effect.Timer
-import org.http4s.client.blaze.BlazeClientBuilder
+import cats.effect.{IO, Timer}
 import cats.syntax.all._
+import cats.instances.list._
+import constellation.util.testing.FakeNode
 import io.circe.Json
-import org.http4s.client.dsl.io._
-import cats.effect.IO
-import constellation.util.testing
+import io.circe.syntax._
+import org.constellation.NetworkLoadbalancer
+import org.constellation.primitives.node.{Id, Info, NodeState}
 import org.http4s.Method._
-import org.http4s.client.blaze._
+import org.http4s.circe.CirceEntityCodec._
 import org.http4s.circe._
-import org.http4s.Uri._
-import io.circe.syntax._
+import org.http4s.client.blaze.BlazeClientBuilder
+import org.http4s.implicits.http4sLiteralsSyntax
 import org.http4s.{Header, Headers, Request, Status}
+import org.scalatest._
+import org.scalatest.time.{Millis, Seconds, Span}
 
+import java.net.InetAddress
 import scala.concurrent.duration._
 import scala.language.postfixOps
-import org.http4s.circe.CirceEntityCodec._
-import java.util.UUID
 
-class IntegrationSpec extends FunSpec with Matchers with BeforeAndAfterAll {
+class IntegrationSpec extends FunSpec with Matchers with BeforeAndAfterAll with SequentialNestedSuiteExecution {
 
   implicit val cs               = IO.contextShift(scala.concurrent.ExecutionContext.global)
   implicit val timer: Timer[IO] = IO.timer(scala.concurrent.ExecutionContext.global)
 
-  val httpClient = BlazeClientBuilder[IO](scala.concurrent.ExecutionContext.global).resource
-
-  val fakeSetup = NonEmptyList.of(
-    Info(Id("node-1"), InetAddress.getLoopbackAddress, 9997, 9998, 123L, NodeState.Ready),
-    Info(Id("node-2"), InetAddress.getLoopbackAddress, 9999, 10000, 456L, NodeState.Ready)
+  val fakeSetup = List(
+    Info(Id("node-1"), InetAddress.getLoopbackAddress, 9997, 9998, 99L, 123L, NodeState.Ready),
+    Info(Id("node-2"), InetAddress.getLoopbackAddress, 9999, 10000, 99L, 456L, NodeState.Ready)
   )
 
-  val clusterInit = fakeSetup.map(i => s"localhost:${i.publicPort}").toList
 
-  val nodes =
-    fakeSetup
-      .map(i => new FakeNode(i.publicPort, i.p2pPort, fakeSetup.toList, Json.obj("node-id" -> i.id.hex.asJson)))
-      .map(_.run)
-      .parSequence
+  def nodesServices(nodesInfo: List[Info]) =
+    nodesInfo
+      .map{n => println(s"Starting for $n with $nodesInfo"); n}
```

Comment:
`println` is a side effect. you can use `Console` from cats and use `flatTap` operator

Reply:
leftover code, is not needed, I'll remove it.